source/Editor.js

```diff
@@ -2085,10 +2090,63 @@ var keyHandlers = {
 
         self.setSelection( range );
     },
-    left: function ( self ) {
+    left: function ( self, event, range) {
+        var selection = self._doc.getSelection();
+
+        // We're not interested in things that are not Carets.
+        if (!selection.isCollapsed) { return; }
+        // or anything that doesn't have an anchorNode.
+        if(!selection.anchorNode) { return; }
+
+        var anchorNode = selection.anchorNode;
+        var anchorPreviousNode = anchorNode.previousSibling;
+
+        if (selection.anchorOffset === 0 &&
+            anchorNode &&
+            anchorPreviousNode && 
+            /SPAN/i.test(anchorPreviousNode.nodeName) && 
+            anchorPreviousNode.classList.contains("mathjax")
+        ) {
+            // this is a mathjax equation, prevent defualt.
+            event.preventDefault();
+
+            // create a new range
+            var leftRange = document.createRange();
+            leftRange.setStartBefore(anchorPreviousNode);
+            leftRange.setEndBefore(anchorPreviousNode);
+            
+            // set a selection.
+            self.setSelection(leftRange);
+        }
+
         self._removeZWS();
     },
-    right: function ( self ) {
+    right: function ( self, event, range) {
+        var selection = self._doc.getSelection();
+
+        // We're not interested in things that are not Carets.
+        if (!selection.isCollapsed) { return; }
+
+        var anchorNode = selection.anchorNode;
+        var anchorNextNode = anchorNode.nextSibling;
+
+        if (
+            anchorNextNode && 
+            /SPAN/i.test(anchorNextNode.nodeName) && 
```

Comment:
You know, this kind of test can be a lot neater with https://developer.mozilla.org/en/docs/Web/API/Element/matches

i.e. `anchorNextNode && anchorNextNode.matches('span.mathjax')`

You may need the non-standard version of this `matchesSelector` for IE.